{"name": "TC015 - Testing the Passwords do not match text", "status": "failed", "statusDetails": {"message": "Failed: Element not found within the wait time", "trace": "setup = <selenium.webdriver.edge.webdriver.WebDriver (session=\"457e3534ebdd522b18a99df5754a24b0\")>\n\n    @allure.feature(\"XEED_Ventures_Login_Page\")\n    @allure.title(\"TC015 - Testing the Passwords do not match text\")\n    @allure.description(\"\"\"\n    **Test Type**\n    Functional and cross-browser testing\n    \n    **Test Scenario**\n    Entering the email, new password but entering different password in confirm password field and waiting till the **Passwords do not match text**\n    \n    **Input data**\n    Email: Sk12@gmail.com\n    New Password: cpktnwt\n    Confirm Password: chethan\n    \n    **Test passes if**\n    The **Valid email is required** text is displayed when confirm password field is entered and waited.\n    \"\"\")\n    \n    def test_suite(setup):\n        driver = setup\n        wait = WebDriverWait(driver, 10)\n    \n        with allure.step(\"Login Page is opened\"):\n            driver.get(\"http://157.15.202.244:99/authentication/login\")\n    \n        with allure.step(\"Clicking on 'Forgot Password?' link\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//a[normalize-space()='Forgot Password?']\")))\n            element.click()\n    \n        with allure.step(\"Entering the Email\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@formcontrolname='email']\")))\n            element.click()\n            element.send_keys(\"Sk12@gmail.com\")\n    \n        with allure.step(\"Entering the NewPassword\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@formcontrolname='newPassword']\")))\n            element.click()\n            element.send_keys(\"cpktnwt\")\n    \n        with allure.step(\"Entering the password in Confirm Password field\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@formcontrolname='confirmPassword']\")))\n            element.click()\n            element.send_keys(\"chethan\")\n    \n        with allure.step(\"Validating 'Passwords do not match' error is shown\"):\n            try:\n>               element = wait.until(EC.visibility_of_element_located((By.XPATH, \"//div[@class='error ng-star-inserted']\")))\n\nXeed_Login_Page\\Test_TC014.py:54: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"457e3534ebdd522b18a99df5754a24b0\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x00000211631D1DA0>, message = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Wait until the method returns a value that is not False.\n    \n        Calls the method provided with the driver as an argument until the\n        return value does not evaluate to ``False``.\n    \n        Parameters:\n        -----------\n        method: callable(WebDriver)\n            - A callable object that takes a WebDriver instance as an argument.\n    \n        message: str\n            - Optional message for :exc:`TimeoutException`\n    \n        Return:\n        -------\n        object: T\n            - The result of the last call to `method`\n    \n        Raises:\n        -------\n        TimeoutException\n            - If 'method' does not return a truthy value within the WebDriverWait\n            object's timeout\n    \n        Example:\n        --------\n        >>> from selenium.webdriver.common.by import By\n        >>> from selenium.webdriver.support.ui import WebDriverWait\n        >>> from selenium.webdriver.support import expected_conditions as EC\n    \n        # Wait until an element is visible on the page\n        >>> wait = WebDriverWait(driver, 10)\n        >>> element = wait.until(EC.visibility_of_element_located((By.ID, \"exampleId\")))\n        >>> print(element.text)\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       \tGetHandleVerifier [0x0x7ff7f8e23865+25605]\nE       \t(No symbol) [0x0x7ff7f8d73970]\nE       \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff7f90a6e7a+1962506]\nE       \t(No symbol) [0x0x7ff7f8b886c4]\nE       \t(No symbol) [0x0x7ff7f8b8898b]\nE       \t(No symbol) [0x0x7ff7f8bc9af7]\nE       \t(No symbol) [0x0x7ff7f8ba9e8f]\nE       \t(No symbol) [0x0x7ff7f8b7ea5d]\nE       \t(No symbol) [0x0x7ff7f8bc75df]\nE       \t(No symbol) [0x0x7ff7f8ba9bb3]\nE       \t(No symbol) [0x0x7ff7f8b7df56]\nE       \t(No symbol) [0x0x7ff7f8b7d463]\nE       \t(No symbol) [0x0x7ff7f8b7dd83]\nE       \t(No symbol) [0x0x7ff7f8c7e10d]\nE       \t(No symbol) [0x0x7ff7f8c8dce8]\nE       \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff7f8f08839+265161]\nE       \tMicrosoft::Applications::Events::EventProperty::to_string [0x0x7ff7f8f10111+296097]\nE       \t(No symbol) [0x0x7ff7f8d82111]\nE       \t(No symbol) [0x0x7ff7f8d7a5b4]\nE       \t(No symbol) [0x0x7ff7f8d7a703]\nE       \t(No symbol) [0x0x7ff7f8d6b8d6]\nE       \tBaseThreadInitThunk [0x0x7ff920ece8d7+23]\nE       \tRtlUserThreadStart [0x0x7ff92209c34c+44]\n\nvenv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:146: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nsetup = <selenium.webdriver.edge.webdriver.WebDriver (session=\"457e3534ebdd522b18a99df5754a24b0\")>\n\n    @allure.feature(\"XEED_Ventures_Login_Page\")\n    @allure.title(\"TC015 - Testing the Passwords do not match text\")\n    @allure.description(\"\"\"\n    **Test Type**\n    Functional and cross-browser testing\n    \n    **Test Scenario**\n    Entering the email, new password but entering different password in confirm password field and waiting till the **Passwords do not match text**\n    \n    **Input data**\n    Email: Sk12@gmail.com\n    New Password: cpktnwt\n    Confirm Password: chethan\n    \n    **Test passes if**\n    The **Valid email is required** text is displayed when confirm password field is entered and waited.\n    \"\"\")\n    \n    def test_suite(setup):\n        driver = setup\n        wait = WebDriverWait(driver, 10)\n    \n        with allure.step(\"Login Page is opened\"):\n            driver.get(\"http://157.15.202.244:99/authentication/login\")\n    \n        with allure.step(\"Clicking on 'Forgot Password?' link\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//a[normalize-space()='Forgot Password?']\")))\n            element.click()\n    \n        with allure.step(\"Entering the Email\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@formcontrolname='email']\")))\n            element.click()\n            element.send_keys(\"Sk12@gmail.com\")\n    \n        with allure.step(\"Entering the NewPassword\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@formcontrolname='newPassword']\")))\n            element.click()\n            element.send_keys(\"cpktnwt\")\n    \n        with allure.step(\"Entering the password in Confirm Password field\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@formcontrolname='confirmPassword']\")))\n            element.click()\n            element.send_keys(\"chethan\")\n    \n        with allure.step(\"Validating 'Passwords do not match' error is shown\"):\n            try:\n                element = wait.until(EC.visibility_of_element_located((By.XPATH, \"//div[@class='error ng-star-inserted']\")))\n                assert element.is_displayed(), \"Expected 'Passwords do not match' message not visible\"\n                allure.attach(element.text, name= \"Text displayed\", attachment_type=allure.attachment_type.TEXT )\n    \n            except:\n>               pytest.fail(\"Element not found within the wait time\")\nE               Failed: Element not found within the wait time\n\nXeed_Login_Page\\Test_TC014.py:59: Failed"}, "description": "\n**Test Type**  \nFunctional and cross-browser testing\n\n**Test Scenario**\nEntering the email, new password but entering different password in confirm password field and waiting till the **Passwords do not match text**\n\n**Input data**\nEmail: Sk12@gmail.com\nNew Password: cpktnwt\nConfirm Password: chethan\n\n**Test passes if**\nThe **Valid email is required** text is displayed when confirm password field is entered and waited.\n", "steps": [{"name": "Login Page is opened", "status": "passed", "start": 1750152418988, "stop": 1750152420607}, {"name": "Clicking on 'Forgot Password?' link", "status": "passed", "start": 1750152420607, "stop": 1750152420941}, {"name": "Entering the Email", "status": "passed", "start": 1750152420941, "stop": 1750152421688}, {"name": "Entering the NewPassword", "status": "passed", "start": 1750152421688, "stop": 1750152421885}, {"name": "Entering the password in Confirm Password field", "status": "passed", "start": 1750152421885, "stop": 1750152422014}, {"name": "Validating 'Passwords do not match' error is shown", "status": "failed", "statusDetails": {"message": "Failed: Element not found within the wait time\n", "trace": "  File \"C:\\Users\\sagar\\PycharmProjects\\TestingProject\\Xeed_Login_Page\\Test_TC014.py\", line 59, in test_suite\n    pytest.fail(\"Element not found within the wait time\")\n    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\sagar\\PycharmProjects\\TestingProject\\venv\\Lib\\site-packages\\_pytest\\outcomes.py\", line 178, in fail\n    raise Failed(msg=reason, pytrace=pytrace)\n"}, "start": 1750152422014, "stop": 1750152432179}], "parameters": [{"name": "setup", "value": "'edge'"}], "start": 1750152418988, "stop": 1750152432180, "uuid": "6ae4b7d2-abd8-404f-bc60-e3a516d3bfe3", "historyId": "f6ca061f4ccffc101d67bf4aec07c487", "testCaseId": "92a9bd4289bc59d9c63837d1d5e4006a", "fullName": "Xeed_Login_Page.Test_TC014#test_suite", "labels": [{"name": "feature", "value": "XEED_Ventures_Login_Page"}, {"name": "parentSuite", "value": "Xeed_Login_Page"}, {"name": "suite", "value": "Test_TC014"}, {"name": "host", "value": "DESKTOP-0Q1NQ4H"}, {"name": "thread", "value": "30940-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Xeed_Login_Page.Test_TC014"}]}